src/environment/game_state.py

```diff
@@ -1,45 +1,45 @@
-from environment.types import Room, Grid, Position
-from llm.llm_story_generator import SchemaSuspect, SchemaVictim
-from typing import List, Dict
+from typing import Dict
 
+from environment.types import CrimeSceneMap, Position, Room
+from llm.marshmallow.schemas.story import StorySchema
+from llm.marshmallow.schemas.suspect import SuspectSchema
+from llm.marshmallow.schemas.victim import VictimSchema
 
 class GameState:
-    def __init__(self,current_location: Room, grid: Grid, dungeon_size: int):
-        self._current_room = current_location
-        self._previous_room = None
-        self._grid = grid
-        self._dungeon_size = dungeon_size
+    def __init__(self, crime_scene_map: CrimeSceneMap, story: StorySchema) -> None:
+        self._crime_scene_map = crime_scene_map
+        self._current_room, self._previous_room = crime_scene_map.start, None
+        
+        self._import_story(story)
 
     @property
-    def victim(self) -> SchemaVictim:
-        return self._victim
-    
-    @victim.setter
-    def victim(self, victim: SchemaVictim):
-        self._victim = victim
-    
-    @property
-    def suspects(self) ->Dict[Position, SchemaSuspect]:
-        return self._suspects
-    
-    @suspects.setter
-    def suspects(self, suspects: Dict[Position, SchemaSuspect]):
-        self._suspects = suspects
+    def crime_scene_map(self) -> CrimeSceneMap:
+        return self.crime_scene_map
 
     @property
     def current_room(self) -> Room:
         return self._current_room
-    
-    @property
-    def dungeon_size(self) -> int:
-        return self._dungeon_size
-    
-    @property
-    def grid(self) -> Grid:
-        return self._grid
-    
+
     @current_room.setter
-    def current_room(self, new_room: Room):
+    def current_room(self, new_room: Room) -> None:
         self._previous_room = self._current_room
         self._current_room = new_room
-    
+
+    @property
+    def suspects(self) -> Dict[Position, SuspectSchema]:
+        return self._suspects
+
+    @property
+    def victim(self) -> VictimSchema:
+        return self._victim
+
+    def _import_story(self, story: StorySchema) -> None:
+        self._victim = story.victim
+        
+        self._suspects = {}
+        for suspect, position in zip(story.suspects, story.suspects_positions):
+            self._suspects[Position(position.row, position.col)] = suspect
+        
+        for room in story.rooms:
+            self._crime_scene_map.rooms[room.row][room.col].name = room.name
+            self._crime_scene_map.rooms[room.row][room.col].description = room.description
```

Comment:
Toto podla mna patri do gamu, pretoze to v buducnosti moze obsahovat nejaku logiku a game_state je hlavne dataclass. Co si o tom myslis?